Drivers/Src/Minions.c

```diff
@@ -1,406 +1,27 @@
-
-    OSMutexPost(
-        &CommMutex,
-        OS_OPT_POST_NONE,
-        &err
-    );
-    assertOSError(0,err);
-
-    //Read Ignition Switch 1
-    uint8_t ign1 = !BSP_GPIO_Read_Pin(PORTA, GPIO_Pin_1);
-
-    //Read Ignition Switch 2
-    uint8_t ign2 = !BSP_GPIO_Read_Pin(PORTA, GPIO_Pin_0);
-    
-    // Seperate the switch bitmap out, so each switch is a byte
-    switchData[10] = ign2;
-    switchData[ 9] = ign1;
-    switchData[ 8] = (SwitchDataReg2 & 0x40) >> 6;
-    for (int i=0; i<8; i++) switchData[i] = (SwitchDataReg1 >> i) & 0x1;
-
-    // Submit the values to the filter, overwrite with stable values
-    SwitchesFilter_put(&SwitchFilter, switchData);
-    SwitchesFilter_get(&SwitchFilter, switchData);
-
-    // Overwrite the old packed values
-    SwitchDataReg1 = 0;
-    for (int i=0; i<8; i++) SwitchDataReg1 |= (switchData[i] << i);
-    SwitchDataReg2 = switchData[8];
-    ign1 = switchData[ 9];
-    ign2 = switchData[10];
-    switchStatesBitmap = (ign2 << 10) | (ign1 << 9) | (SwitchDataReg2 << 8) | (SwitchDataReg1);
-}
-
-/**
- * @brief Toggles a light. Should be used only after Lights_Toggle_Set has been called for this light so that we are accurately tracking the enabled and disabled lights
- * @param light Which light to toggle
-*/
-void Lights_Toggle(light_t light){
-    State lightState = Lights_Read(light);
-    if(lightState == OFF){
-        Lights_Set(light, ON);
-    } else {
-        Lights_Set(light, OFF);
-    }
-}
-
-/**
-
- * @brief   Set light toggling
- * @param   light Which light to enable/disable toggling for
- * @param   state State to set toggling (ON/OFF)
- * @return  void
- */
-void Lights_Toggle_Set(light_t light, State state) {
-    // Mutex not needed here because only BlinkLights uses this bitmap
-    if(light == LEFT_BLINK){
-        (state == ON) ? (lightToggleBitmap |= 0x02) : (lightToggleBitmap &= 0xFD);
-    }
-    else if(light == RIGHT_BLINK){
-        (state == ON) ? (lightToggleBitmap |= 0x01) : (lightToggleBitmap &= 0xFE);
-    }
-
-}
 
-/**
-* @brief   Read the state of a specific toggleable light from the toggle bitmap
-* @param   light Which Light to read
-* @return  returns State enum which indicates ON/OFF
-*/
-State Lights_Toggle_Read(light_t light) {
-    if(light == LEFT_BLINK){
-        return (State) ((lightToggleBitmap & 0x02) >> 1);
-    }
-    else if(light == RIGHT_BLINK){
-        return (State) (lightToggleBitmap & 0x01);
-    }
-    else{
-        return OFF;
+void Minion_Init(void){
+    for(uint8_t i = 0; i < MINIONPIN_NUM; i++){
+        BSP_GPIO_Init(PINS_LOOKARR[i].pinmask, PINS_LOOKARR[i].port, PINS_LOOKARR[i].direction);
     }
 }
 
-/**
- * @brief   Read toggle bitmap
- * @return  returns uint8_t bitmap for toggle (LEFT/RIGHT)
- */
-uint8_t Lights_Toggle_Bitmap_Read(void) {
-    return lightToggleBitmap;
+bool Minion_Read_Switch(MinionPin_t pin){
+    if(pin == BRAKELIGHT){
+        return false; //trying to read from an output pin, can't do that.
+    }    
```

Comment:
This isn't generic behavior. This error check should lookup the passed-in pin in the lookup table, and use the returned PinInfo_t struct to determine if the pin is an output. This way, if we add more outputs in the future, we won't have to revisit this.